### ISSUE TITLE
developer docs broken links / api folder missing

### ISSUE THREAD

**Issue by worldpeace-germany** [missionpinball/mpf#1689]
Hello,

my assumption would be that in https://github.com/missionpinball/mpf/tree/dev/docs I should see a folder "api" holding the api developer documentation. It is not there, though the index.rst I believes indicates that I  should look there. 

If I use the "edit on github" link on the web page, e.g. https://developer.missionpinball.org/en/dev/api/self.machine.events.html  I am  routed to https://github.com/missionpinball/mpf/blob/dev/docs/api/self.machine.events.rst again a folder called api should be present. I am getting a 404 error.

I would like to contribute to the documentation, can someone point me to the right direction or fix the problem if it is not just me missing something?

Thanks.

**Comment by toomanybrians** [comment]
Good question. This highlights the need for some type of readme which explains how the docs are built.

The contents of the `/docs/api` folder are built dynamically from docstrings in the source code. You can look at [`/docs/conf.py` line 365](https://github.com/missionpinball/mpf/blob/dev/docs/conf.py#365) to see the code that does this.

The templates for this are in the [`/docs/_templates`](https://github.com/missionpinball/mpf/tree/dev/docs/_templates) folder.

So depending on what you want to edit, you'll most likely edit it at the source python file. (The idea was it's easier to keep everything up to date if the docstrings are being created as part of the code.)

That said, I haven't even really thought about the developer documentation since I've gotten back involved with this project last year. Looks like it's several versions old and needs to be setup to run for the newer versions. I'm not sure what that would involve, if you want to take a look I can see how I can help.

**Comment by worldpeace-germany** [comment]
Thanks for the answer, that gives some initial help. I have created a pull request. Let's see once it is accepted how that changes the  developer documentation.

Don't get me wrong, I don't want to start a discussion just want to point out my opinion and of course I value other opinions. From my point of view (might be due to the fact that I am not a hard core python developer) this approach has some downsides. I believe there is no way that people have to much documentation, hence the ability to change documentation should have a low entry barrier. Changing  documentation in code might scare some people. I was able to add my piece, but still haven't figured out how to create for example a fully new section if that would be needed. In addition I have no idea how to generate the newly written documentation to test it. Which might of course cause some additional work on the maintainer in case something is wrong in my changes.  Last but not least the false error messages and  links in the documentation I pointed already out in the OP, that throws people off the first time and might stop them to contribute.

As I said only my 5ct. Thanks for your support, I have a basic idea what I have to do in the future that is good enough for now for me.

**Comment by toomanybrians** [comment]
I agree that this is confusing. I started a different thread last week to discuss the topic of the MPF websites & docs: https://github.com/orgs/missionpinball/discussions/100

TLDR is I agree with you, the docs need to be simple to encourage others to contribute. I want to modernize and simplify everything and that will include getting the developer docs to move to a modern “expected” behavior too.
